libs/logic-apps-shared/src/utils/src/lib/helpers/__test__/connectors.spec.tsx

```diff
@@ -210,4 +212,50 @@ describe('utils/src/lib/helpers/connectors', () => {
       expect(getDescriptionFromConnector(undefined)).toBe('');
     });
   });
+
+  describe('normalizeConnectorId', () => {
+    const armConnectorId = '/subscriptions/#subscription#/providers/Microsoft.Web/locations/#location#/managedApis/sql';
+    const armConnectorId2 = '/subscriptions/#subscription#/providers/Microsoft.Web/locations/#location#/managedApis/gmail';
+    const armConnectorIds = [armConnectorId, armConnectorId2];
+    const subscriptionId = '00000000-0000-0000-0000-000000000000';
+    const location = 'eastus';
+
+    it('should replace subscriptionId and location correctly in arm connector id', async () => {
+      const expectedConnectorId = `/subscriptions/${subscriptionId}/providers/Microsoft.Web/locations/${location}/managedApis/sql`;
+      expect(normalizeConnectorId(armConnectorId, subscriptionId, location)).toEqual(expectedConnectorId);
+
+      expect(normalizeConnectorId(armConnectorId, '', '')).toEqual('/subscriptions//providers/Microsoft.Web/locations//managedApis/sql');
+    });
+
+    it('should not change connectorId when not an arm resource', async () => {
+      expect(normalizeConnectorId('', subscriptionId, location)).toEqual('');
+      expect(normalizeConnectorId('/serviceProviders/sql', subscriptionId, location)).toEqual('/serviceProviders/sql');
+      expect(normalizeConnectorId('/dataOperations', '', '')).toEqual('/dataOperations');
+    });
+
+    it('check casing', async () => {
+      expect(normalizeConnectorId('Hello', subscriptionId, location)).not.toEqual('hello');
+      expect(normalizeConnectorId('Hello', subscriptionId, location, true)).toEqual('hello');
+      expect(normalizeConnectorId('/serviceProviders/sql', subscriptionId, location)).not.toEqual('/serviceProviders/SQL'.toLowerCase());
+      expect(normalizeConnectorId('/serviceProviders/sql', subscriptionId, location, true)).toEqual('/serviceProviders/SQL'.toLowerCase());
+    });
+
+    it('should replace all subscriptionIds and locations correctly in arm connector ids only', async () => {
+      const expectedConnectorId = `/subscriptions/${subscriptionId}/providers/Microsoft.Web/locations/${location}/managedApis/sql`;
+      const expectedConnectorId2 = `/subscriptions/${subscriptionId}/providers/Microsoft.Web/locations/${location}/managedApis/gmail`;
+
+      expect(normalizeConnectorIds([...armConnectorIds, '', '/serviceProviders/sql'], subscriptionId, location)).not.toEqual([
+        expectedConnectorId,
+        expectedConnectorId2,
+        '',
+        '/serviceProviders/Sql',
+      ]);
+      expect(normalizeConnectorIds([...armConnectorIds, '', '/serviceProviders/sql'], subscriptionId, location, true)).toEqual([
+        expectedConnectorId.toLowerCase(),
+        expectedConnectorId2.toLowerCase(),
+        '',
+        '/serviceProviders/Sql'.toLowerCase(),
+      ]);
+    });
```

Comment:
This should be in a separate describe since this is a difference function normalizeConnectorIds